apps/system/test/raptor/lockscreen.esp.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
+
+/* globals require, module, Performance */
+
+module.exports = function(filepath, outpath, requiredir) {
+  var Logger = function() {
+    this._writelogs = {};
+  };
+  Logger.prototype.log = function(outpath, advice) {
+    this._writelogs[filepath] = {
+      'outpath': outpath,
+      'file': advice.file,
+      'query': advice.query
+    };
+  };
+  var logger = new Logger();
+  var opts = {};
+  if (outpath) {
+    var fs = require('fs');
+    opts.writer = {
+      write: function(advice) {
+        if (advice.instance.applied) {
+          logger.log(filepath, advice);
+        }
+        var code = advice.code;
+        fs.writeFile(outpath, code);
+      }
+    };
+  }
+  var Espect = require(requiredir + 'espect.js');
+  var espect = new Espect(opts);
+  espect
+    .select(filepath + ' LockScreen.prototype.unlock')
+    .before(function() {
+      Performance.mark('LockScreen#unlock');
+    })
+    .done()
+  .done();
```

Comment:
It's an API from Espect: the first `done` is for the `select`, and the second one is for the whole process constructed here. It's weird only when you select single file to query. If you have multiple selection in one process it would be like:

    espect()
       select()
       ....
       .done()
       .select()
       .done()
    .done()